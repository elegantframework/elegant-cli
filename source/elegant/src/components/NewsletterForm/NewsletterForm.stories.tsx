import type { Meta, StoryObj } from '@storybook/react';
import { NewsletterForm } from './NewsletterForm';

const meta: Meta<typeof NewsletterForm> = {
  title: 'Core/Forms/Newsletter Form',
  component: NewsletterForm,
};

export default meta;

type Story = StoryObj<typeof NewsletterForm>;

export const Form: Story = {
  args: {},
};