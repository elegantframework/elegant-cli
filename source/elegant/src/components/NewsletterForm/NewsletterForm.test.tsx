import React from "react";
import renderer from 'react-test-renderer';
import { NewsletterForm } from "./NewsletterForm";

describe("Newsletter Signup Form", () => {
    it('handles an action', () => {
        const form = renderer
        .create(<NewsletterForm action={() => {console.log('hello world')}}/>)
        .toJSON();
        expect(form).toMatchSnapshot();
    });
});