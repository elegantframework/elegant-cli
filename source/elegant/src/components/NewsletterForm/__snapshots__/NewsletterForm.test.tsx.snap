// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Newsletter Signup Form handles an action 1`] = `
<form
  action={[Function]}
  className="flex flex-wrap -mx-2"
  method="post"
  target="_blank"
>
  <div
    className="px-2 grow-[9999] basis-64 mt-3"
  >
    <div
      className="group relative"
    >
      <svg
        aria-hidden="true"
        className="w-6 h-full absolute inset-y-0 left-3 text-slate-400 pointer-events-none group-focus-within:text-primary-500 dark:group-focus-within:text-slate-400"
        fill="none"
        stroke="currentColor"
        strokeLinecap="round"
        strokeLinejoin="round"
        strokeWidth="2"
        viewBox="0 0 24 24"
      >
        <path
          d="M5 7.92C5 6.86 5.865 6 6.931 6h10.138C18.135 6 19 6.86 19 7.92v8.16c0 1.06-.865 1.92-1.931 1.92H6.931A1.926 1.926 0 0 1 5 16.08V7.92Z"
        />
        <path
          d="m6 7 6 5 6-5"
        />
      </svg>
      <input
        aria-label="Email address"
        autoComplete="email"
        className="appearance-none shadow rounded-md ring-1 ring-slate-900/5 leading-5 sm:text-sm border border-transparent py-2 placeholder:text-slate-400 pl-12 pr-3 block w-full text-slate-900 focus:outline-none focus:ring-2 focus:ring-primary-500 bg-white dark:bg-slate-700/20 dark:ring-slate-200/20 dark:focus:ring-primary-500 dark:text-white"
        name="email_address"
        placeholder="Subscribe via email"
        required={true}
        type="email"
      />
    </div>
  </div>
  <div
    className="px-2 grow flex mt-3"
  >
    <button
      className="bg-primary-500 flex-auto shadow text-white rounded-md text-sm border-y border-transparent py-2 font-semibold px-3 hover:bg-primary-600 dark:hover:bg-primary-400 focus:outline-none focus:ring-2 focus:ring-offset-2 focus:ring-primary-300 dark:focus:ring-offset-slate-900 dark:focus:ring-primary-700"
      type="submit"
    >
      Subscribe
    </button>
  </div>
</form>
`;
